crates/turborepo-discovery/src/lib.rs

```diff
@@ -0,0 +1,118 @@
+//! turborepo-discovery
+//!
+//! This package contains a number of strategies for discovering various things
+//! about a workspace. These traits come with a basic implementation and some
+//! adaptors that can be used to compose them together.
+//!
+//! This powers various intents such as 'query the daemon for this data, or
+//! fallback to local discovery if the daemon is not available'. Eventually,
+//! these strategies will implement some sort of monad-style composition so that
+//! we can track areas of run that are performing sub-optimally.
+
+#![feature(async_fn_in_trait)]
+#![feature(return_position_impl_trait_in_trait)]
+
+use tokio_stream::{iter, StreamExt};
+use turbopath::AbsoluteSystemPathBuf;
+use turborepo_repository::package_manager::{self, PackageManager};
+
+#[derive(Clone)]
+pub struct RootWorkspaceData {
```

Comment:
How do you feel about including `PackageManager` as part of this data? I think it'll be very hard to avoid performing detection twice if we don't shore this information up here. `PackageGraph` will always need to know `PackageManager` even if we can get all of the workspaces discovered via the daemon we would still need to `PackageManager::get_package_manager`.

I think it makes sense that if workspaces haven't changed that means the package manager hasn't either.

Also, I'm so sorry, but I realized that this isn't root workspace and just workspace data 🤦

Reply:
I think package manager makes sense.